WordPress/Classes/Services/PostAutoUploadInteractor.swift

```diff
@@ -43,7 +43,8 @@ final class PostAutoUploadInteractor {
         if post.isLocalDraft || post.shouldAttemptAutoUpload {
             return .upload
         } else {
-            return .autoSave
+            // because autosave call will end up uplaoding a post on self hosted we don't autosave in this case
+            return post.isHostedAtWPcom ? .autoSave : .nothing
```

Comment:
`post.isHostedAtWPcom` will return `false` for AT sites.
In `develop` I'm able to `autoSave` content for an AT site. However, after this change, this doesn't happen anymore.

I did some investigation here and this seems to solve the issue:

```swift
return post.blog.supports(.wpComRESTAPI) ? .autoSave : .nothing
```

Would you mind to update the code?

The steps to reproduce the issue:

1. Use an AT site
2. While offline, edit a draft and then tap cancel
3. Go back online
4. The post is auto-saved
5. If I go to calypso I see an "A more recent revision of this post exists. Restore?" message

Reply:
Excellent! Updating